### PR TITLE
[PREGEL] Add storing state to conductor state machine

### PR DESCRIPTION
**Pull request by jvolmer**
Second PR for adding a state machine to the pregel conductor. 
Adds storing state to conductor state machine

### CODE REVIEW

**Comment by jvolmer** [comment]
TODO merge parent first

File: arangod/Pregel/Conductor.cpp [hunk 1]
```diff
@@ -261,10 +262,14 @@ bool Conductor::_startGlobalStep() {
...
-      _feature.metrics()->pregelConductorsStoringNumber->fetch_add(1);
-      _timing.storing.start();
-      _finalizeWorkers();
+      if (_masterContext) {
+        _masterContext->postApplication();
+      }
+      RecoveryManager* mngr = _feature.recoveryManager();
+      if (mngr) {
+        mngr->stopMonitoring(this);
+      }
```
Comment (jvolmer) [inline arangod/Pregel/Conductor.cpp:271]: First half of the _finalizeWorkers() method

File: arangod/Pregel/Conductor/StoringState.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  LOG_PREGEL("fc187", DEBUG) << "Finalizing workers";
+
+  auto finalizeExecutionCommand =
+      FinalizeExecution{.executionNumber = conductor._executionNumber,
+                        .gss = conductor._globalSuperstep,
+                        .withStoring = true};
+  VPackBuilder command;
+  serialize(command, finalizeExecutionCommand);
+
+  conductor._sendToAllDBServers(Utils::finalizeExecutionPath, command);
```
Comment (jvolmer) [inline arangod/Pregel/Conductor/StoringState.cpp:40]: Second half of the _finalizeWorkers() method

**Review by markuspf — APPROVED** [review]
LGTM

**Pull request merged**